source/D2Game/src/GAME/Game.cpp

```diff
@@ -2620,192 +2419,74 @@ D2GameStrc* __fastcall GAME_GetGameByClientId(int32_t nClientId)
-//                                     + 4);
-//                if ((int32_t)v5 <= 0)
-//                {
-//                    LeaveCriticalSection(&v1->pLock);
-//                    return;
-//                }
-//            }
-//            if (v5)
-//            {
-//                if (gpD2ServerCallbackFunctions_6FD45830 && gpD2ServerCallbackFunctions_6FD45830->pfCloseGame)
-//                {
-//                    if (IsBadCodePtr((FARPROC)gpD2ServerCallbackFunctions_6FD45830->pfCloseGame))
-//                    {
-//                        FOG_DisplayAssert("ptEventCallbackTable->fpCloseGame", __FILE__, __LINE__);
-//                        exit(-1);
-//                    }
-//                    gpD2ServerCallbackFunctions_6FD45830->pfCloseGame(v5->nServerToken, v5->bExpansion != 0 ? 0x44325850u : 0x44324456u, v5->dwSpawnedPlayers, (signed __int32_t32)v5->dwGameFrame / 25);
-//                }
-//                v6 = v5->lpCriticalSection;
-//                if (!v6)
-//                {
-//                    FOG_DisplayAssert("ptGame->_ptLock != nullptr", __FILE__, __LINE__);
-//                    exit(-1);
-//                }
-//                EnterCriticalSection(v6);
-//                GAME_FreeGame(v2, v5);
-//                v7 = gpGameDataTbl_6FD45818;
-//                if (v5->__0000[2])
-//                {
-//                    v8 = (char*)&v5->__0000[1];
-//                    if (v5->__0000[1])
-//                    {
-//                        *(_DWORD*)sub_6FC3B560(&v5->__0000[1], -1) = *(_DWORD*)v8;
-//                        *(_DWORD*)(*(_DWORD*)v8 + 4) = v5->__0000[2];
-//                        *(_DWORD*)v8 = 0;
-//                        v5->__0000[2] = 0;
-//                    }
-//                    v9 = (char*)&v5->__0000[3];
-//                    if (v5->__0000[3])
-//                    {
-//                        *(_DWORD*)sub_6FC3B560(&v5->__0000[3], -1) = *(_DWORD*)v9;
-//                        *(_DWORD*)(*(_DWORD*)v9 + 4) = v5->pNext;
-//                        *(_DWORD*)v9 = 0;
-//                        v5->pNext = 0;
-//                    }
-//                }
-//                (*(void(__thiscall**)(D2GameManagerStrc*, D2GameStrc*))v7->field_0.field_0)(v7, v5);
-//                v3 = &v7->pLock;
-//            }
-//        }
-//        LeaveCriticalSection(v3);
-//    }
+    if (gpGameDataTbl_6FD45818)
+    {
+        GAMEDATALOCKEDHANDLE hLock;
+        if (D2GameStrc* pGame = gpGameDataTbl_6FD45818->tHashTable.Lock(GetGameHandleFromHashValue(nGameGUID), &hLock, TRUE))
+        {
+            if (gpD2EventCallbackTable_6FD45830 && gpD2EventCallbackTable_6FD45830->pfCloseGame)
+            {
+                D2_ASSERT(!IsBadCodePtr((FARPROC)gpD2EventCallbackTable_6FD45830->pfCloseGame));
+                gpD2EventCallbackTable_6FD45830->pfCloseGame(
+                    pGame->nServerToken,
+                    pGame->bExpansion ? 0x44325850 : 0x44324456,
```

Comment:
These two constants are ASCII literals of the product codes, `'D2XP'` &  `'D2DV'`respectively